### PR TITLE
fix: dashboard/onboarding polish — tour banners, compact collisions, subtab clip, badge aging (#1067, #1098, #1099, #708)

### LINKED ISSUE #708: [Office][Bug] Badge old-item tint ignores stale time-off/tool/deletion approvals
Status: OPEN

## Bug Description
`BadgeCountService.BadgeCounts.hasOldItems` is documented as "True if any item has been pending for more than 7 days", but `BadgeCountService.getAllBadgeCounts(userId:)` only computes `oldestPendingDate` from pending/in-review JPOs.

That means the app can show a non-red / non-aged badge state even when the Office tab has old pending time-off requests, pending tool edit verifications, or pending deletion approvals. The badge number includes those queues, but the age/tint signal ignores them.

## Evidence
Current source:

- `core/Sources/WiredPartCore/Services/BadgeCountService.swift:78-82` says `hasOldItems` should be true when any item has been pending more than 7 days.
- `core/Sources/WiredPartCore/Services/BadgeCountService.swift:108-109` includes `pendingTimeOff + pendingToolEdits + pendingDeletions` in the Office badge.
- `core/Sources/WiredPartCore/Services/BadgeCountService.swift:181-197` counts time-off, tool-edit, and deletion pending queues.
- `core/Sources/WiredPartCore/Services/BadgeCountService.swift:232-236` sets `oldestPendingDate` from only:

```sql
SELECT MIN(created_at) FROM job_parts_orders
WHERE status IN ('pending', 'in_review') AND deleted_at IS NULL
```

So old non-JPO office work is counted in badges but invisible to `hasOldItems`.

## Steps to Reproduce
1. On a fresh DB, insert a `schedule_exceptions` time-off row with `exception_type = 'time_off'`, `is_approved = 0`, `deleted_at IS NULL`, and an old creation/date field if available.
2. Ensure there are no `job_parts_orders` rows with `status IN ('pending', 'in_review')`.
3. Call `BadgeCountService(db: db).getAllBadgeCounts(userId: someUser)`.
4. Observe `counts.officeBadge > 0` but `counts.oldestPendingDate == nil` and `counts.hasOldItems == false`.

The same gap applies to old `tool_change_log.verification_status = 'pending'` rows and old `scheduled_deletions.status = 'pending_approval'` rows.

## Expected Behavior
`oldestPendingDate` / `hasOldItems` should consider every pending queue that contributes to user-visible badge urgency, at minimum:

- Pending/in-review JPOs (`job_parts_orders.created_at`)
- Pending time-off requests (`schedule_exceptions` date/created timestamp)
- Pending tool edit verifications (`tool_change_log.created_at` or equivalent)
- Pending deletion approvals (`scheduled_deletions.created_at` or equivalent)

## Actual Behavior
Only JPO age is considered. Old Office approvals outside orders can look fresh/non-urgent even while the Office badge count is non-zero.

## Impact / Severity
Severity: Medium

This is a workflow-priority bug: managers can miss stale approvals because the numeric Office badge and badge urgency/tint disagree. It undermines the "old pending items turn red" signal described by the service itself.

## Suggested Fix
Compute `oldestPendingDate` as the minimum across all pending badge queues (using each table's best available timestamp), or track per-module oldest dates and have Office badge tint derive from the Office queues specifically. Add regression tests for:

- Old pending time-off with no pending JPO => `hasOldItems == true`
- Old pending tool edit with no pending JPO => `hasOldItems == true`
- Old pending deletion with no pending JPO => `hasOldItems == true`
- Existing pending JPO behavior still works

### LINKED ISSUE #1067: [Onboarding][Bug] Completed walkthrough leaves tour banners active
## Bug Description
Completing or skipping the guided onboarding walkthrough leaves `OnboardingProgressManager.isOnboardingActive` set to `true` instead of ending the active tour state. Because `OnboardingBanner` renders whenever `isOnboardingActive && !tasks.isEmpty`, completed users keep seeing per-page tour banners after they pressed "Go to Dashboard" or "Skip Onboarding".

## Evidence
- `Weird Parts IOS/Weird Parts IOS/Auth/OnboardingWalkthroughView.swift:586-595` calls `finishOnboarding()`, marks `hasCompletedOnboarding = true`, then sets `manager.isOnboardingActive = true`.
- `Weird Parts IOS/Weird Parts IOS/Shared/OnboardingProgressManager.swift:76-80` persists the active flag to `UserDefaults`, so the stale active state survives relaunch.
- `Weird Parts IOS/Weird Parts IOS/Shared/OnboardingBanner.swift:16` shows the banner whenever `onboardingManager.isOnboardingActive && !tasks.isEmpty`, even when all required tasks are complete.

## Steps to Reproduce
1. Start as a new user with an `OnboardingProgressManager` available.
2. Complete or skip modules in the guided walkthrough.
3. Press "Go to Dashboard" (or use "Skip Onboarding" from the welcome screen).
4. Open a page that has registered onboarding tasks, such as dashboard/jobs/parts.

## Expected Behavior
Finishing/skipping the full onboarding walkthrough should set the per-user active tour flag to false and persist it, so regular app pages do not show tour banners unless the user explicitly restarts onboarding.

## Actual Behavior
`finishOnboarding()` persists completion flags but leaves `isOnboardingActive` true, so the page-level banner continues to render completed/incomplete tour prompts after onboarding is done.

## Impact / Severity
Severity: Medium.
Blast radius: first-run and field-test UX across every page with onboarding tasks. It can make users think setup is still in progress after explicitly completing onboarding, adding persistent clutter to high-use inventory/jobs/time workflows.

## Suggested Fix
Change `finishOnboarding()` to set `manager.isOnboardingActive = false` and persist that value (or add a dedicated manager method such as `completeTour()` that marks completed walkthrough tasks and deactivates the active tour). Add a regression test that completion does not leave the active flag true.

## Discovered By
Paperclip WEI-3842 daily exploratory bug hunt.

### LINKED ISSUE #1098: [Dashboard][Bug] Compact onboarding content collides with FAB, toast, and bottom tab bar
Status: OPEN

Found during weekly UI usability verification (Paperclip WEI-3982).

Environment/evidence:
- Branch/worktree: WEI-3982-weekly-ui-usability-verification
- Device: WEI3988-iPhone simulator, iOS 26.5, portrait
- Command: `WEI_936_ARTIFACT_DIR=/tmp/wei-3982-ui-qa/onboarding-phone xcodebuild test -project "Weird Parts IOS/Weird Parts.xcodeproj" -scheme "Weird Parts" -destination 'platform=iOS Simulator,name=WEI3988-iPhone,OS=26.5' -only-testing:"Weird PartsUITests/WEI936OnboardingQAUITests"`
- Result: TEST SUCCEEDED; screenshots exported from xcresult with `xcrun xcresulttool export attachments`
- Evidence screenshots:
  - `wei936-02-card-not-started` exported at `/tmp/wei-3982-ui-qa/xcattachments/onboarding/AEC83E3A-15AE-400C-9C10-E21480345F34.png`
  - `wei936-03-in-progress-banner` exported at `/tmp/wei-3982-ui-qa/xcattachments/onboarding/9EB33C5A-8FD1-4416-B701-A0525D51CEF2.png`
  - `wei936-05-dismiss-toast` exported at `/tmp/wei-3982-ui-qa/xcattachments/onboarding/2D423B67-E3A7-4238-93C0-73B4BB019F10.png`

Steps to reproduce:
1. Run the WEI936 onboarding UI tests on a compact iPhone simulator.
2. Inspect the not-started checklist, in-progress banner, and dismiss-toast states.

Actual:
- The floating blue AI/sparkle button overlaps the lower-right portion of the onboarding/progress card.
- In the dismiss-toast state, the same FAB visually collides with the toast and sits close to the `Undo` action.
- The bottom tab bar overlays lower dashboard/onboarding card content; rows behind the tab bar are blurred/faded and partly unreadable.

Expected:
- Floating actions should avoid toast/snackbar and bottom-tab safe areas.
- Scrollable dashboard/onboarding content should include enough bottom inset so final rows are readable and reachable above the tab bar.

Impact:
- Medium UX/accessibility risk on compact phones: users can miss onboarding steps, and the toast undo affordance competes with the FAB.

Recommended owner/approach:
- Front-end/iOS UI owner.
- Add bottom safe-area/content inset for Dashboard onboarding cards and adjust FAB placement when bottom overlays/toasts are visible. Verify on compact iPhone portrait and iPad/tablet.

### LINKED ISSUE #1099: [Dashboard][Bug] Compact Dashboard subtab row clips trailing item
Status: OPEN

Found during weekly UI usability verification (Paperclip WEI-3982).

Environment/evidence:
- Branch/worktree: WEI-3982-weekly-ui-usability-verification
- Device: WEI3988-iPhone simulator, iOS 26.5, portrait
- Command: `WEI_936_ARTIFACT_DIR=/tmp/wei-3982-ui-qa/onboarding-phone xcodebuild test -project "Weird Parts IOS/Weird Parts.xcodeproj" -scheme "Weird Parts" -destination 'platform=iOS Simulator,name=WEI3988-iPhone,OS=26.5' -only-testing:"Weird PartsUITests/WEI936OnboardingQAUITests"`
- Result: TEST SUCCEEDED; screenshots exported from xcresult with `xcrun xcresulttool export attachments`
- Evidence screenshots:
  - `wei936-02-card-not-started` exported at `/tmp/wei-3982-ui-qa/xcattachments/onboarding/AEC83E3A-15AE-400C-9C10-E21480345F34.png`
  - `wei936-03-in-progress-banner` exported at `/tmp/wei-3982-ui-qa/xcattachments/onboarding/9EB33C5A-8FD1-4416-B701-A0525D51CEF2.png`

Steps to reproduce:
1. Run the WEI936 onboarding UI tests on a compact iPhone simulator.
2. Inspect the Dashboard top horizontal subtab row in the not-started and in-progress onboarding states.

Actual:
- The Dashboard horizontal subtab row is clipped on the right edge after `Daily Report`; the next icon/text is partially cut off at the screen edge.

Expected:
- Horizontal tab rows should either scroll cleanly with visible affordance/padding or wrap/size so no partially clipped tab is presented as broken content.

Impact:
- Low/medium responsive UX polish issue: compact iPhone users see a truncated navigation item and may not understand that more tabs are available.

Recommended owner/approach:
- Front-end/iOS UI owner.
- Add trailing padding/scroll indicator behavior or compact-label handling to the Dashboard subtab strip. Verify on compact iPhone portrait and tablet.

### PR DESCRIPTION
**Pull request by xXKillerNoobYT**
Batch O (dashboard/onboarding polish) per `docs/plans/beta-readiness-issue-resolution-2026-07.md`.

## Per-item changes

- **#1067 — Completed walkthrough leaves tour banners active**
  `OnboardingWalkthroughView.finishOnboarding()` was setting `manager.isOnboardingActive = true` after completing/skipping the walkthrough, so per-page tour banners kept rendering. It now calls a new persisted `OnboardingProgressManager.endTour()`. Added `startTour()`/`endTour()` methods that set **and persist** the active flag, and migrated every direct `isOnboardingActive` assignment to them (NewUserWelcomeView auto-start, DashboardView "Take the Full App Tour" and "End Tour", AppConfigPage "Restart App Tour", UserMenuSheet "Restart setup checklist"). The dashboard "End Tour" button previously set the flag in memory only — banners returned on relaunch; that's the same defect class and is fixed by the persisted method.
- **#1098 — Compact onboarding content collides with FAB, toast, and bottom tab bar**
  Dashboard scroll content now always reserves 76pt bottom clearance (52pt AI FAB + margins, added on top of the existing accessibility-size padding) so the final onboarding/progress card scrolls fully clear of the floating AI button and bottom tab bar. The checklist dismiss toast gets 76pt trailing clearance so its Undo action never sits under the FAB, and Undo gains a `dsMinTapTarget()` 44pt tap target. "End Tour" also gets a 44pt tap frame.
- **#1099 — Compact Dashboard subtab row clips trailing item**
  `ModuleHostView.subTabPicker` (shared by all modules, including the Dashboard strip that clipped after "Daily Report") now tracks scroll-edge state via `onScrollGeometryChange` and shows leading/trailing gradient fade scrims while more chips remain off-screen — a visible scroll affordance so a partially clipped chip reads as scrollable content, not broken layout. Scrims are hit-test transparent and hidden from accessibility. The `WarehouseTabStripRegressionTests` source assertions (ScrollViewReader auto-scroll behavior) are preserved.
- **#708 — Badge old-item tint ignores stale time-off/tool/deletion approvals**
  `BadgeCountService.getAllBadgeCounts` now computes `oldestPendingDate` across **all** pending badge queues: pending/in-review JPOs (`created_at`), pending time-off (`schedule_exceptions.created_at`), pending tool edit verifications (`tool_change_log.changed_at` — that table has no `created_at`), and pending deletion approvals (`scheduled_deletions.created_at`). Each table is queried separately so a missing table on a partially-migrated DB only drops that queue. The overall oldest is chosen by parsed date (queues mix ISO 8601 and SQLite `datetime('now')` formats, which don't sort lexicographically). `hasOldItems` now parses via `CoreFormatters.parseDateTime` so space-format SQLite default timestamps age correctly — previously they silently disabled the aged tint.

## Testing

- `xcrun swiftc -parse` clean on all 9 edited Swift files.
- `cd core && swift build` — Build complete.
- `swift test --filter BadgeCountService` — 24/24 passing (19 existing + 5 new regression tests: stale time-off / tool edit / deletion each drive `hasOldItems` with no JPOs present; JPOs still feed the signal and the overall-oldest queue item wins; space-format timestamp parsing).
- Full core `swift test` run green locally.
- UI-visual verification (compact iPhone portrait FAB/toast/subtab rendering) needs a simulator and lands on the **local Mac runner** Xcode lane; code-level checks (safe-area padding math, 44pt tap targets, accessibility hiding on decorative scrims) verified by inspection.

CI routing: hybrid runners — ubuntu-latest handles the non-Xcode PR gates (Artifact Guard, CodeQL compat, swift-analyze recovery, merge maintenance); the local self-hosted Mac runner (`[self-hosted, macOS, ARM64, xcode, ios, local-mac]`) owns the Xcode build/test/QA path.

Closes #1067
Closes #1098
Closes #1099
Closes #708

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Weird Parts IOS/Weird Parts IOS/Shared/OnboardingProgressManager.swift [hunk 1]
```diff
@@ -35,6 +35,22 @@ class OnboardingProgressManager: ObservableObject {
         saveProgress()
     }
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply Weird Parts IOS/Weird Parts IOS/Shared/OnboardingProgressManager.swift:41]: Fixed in 4d7400dc3. Took the option you suggested: `resetProgress()` now only clears/persists task state and no longer touches `isOnboardingActive` — tour activation is owned exclusively by `startTour()`/`endTour()`. Audited all callers: the only two restart sites (AppConfigPage, UserMenuSheet) already pair `resetProgress()` with `startTour()`, so every call site stays correct with a single publish/persist per flag.

File: core/Sources/WiredPartCore/Services/BadgeCountService.swift [hunk 2]
```diff
@@ -76,9 +76,14 @@ public final class BadgeCountService: Sendable {
...
+        ///
+        /// Uses `parseDateTime` (not `parseISO`) because the pending queues mix
+        /// timestamp formats: service-written rows carry ISO 8601 strings while
+        /// rows created via SQLite column defaults carry `datetime('now')`
+        /// space-separated strings (#708).
         public var hasOldItems: Bool {
             guard let dateStr = oldestPendingDate else { return false }
-            guard let date = CoreFormatters.parseISO(dateStr) else { return false }
+            guard let date = CoreFormatters.parseDateTime(dateStr) else { return false }
             return Date().timeIntervalSince(date) > 7 * 86400
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply core/Sources/WiredPartCore/Services/BadgeCountService.swift:88]: Fixed in 4d7400dc3. Audited all `parseDateTime` callers first: 15+ sites (DashboardService, ChatService, PeopleService, BreakService, DailyReportGenerator, AuthService, ReportsService, IOSTimesheetsPage) parse clock-in/out and correction timestamps whose local-time interpretation could not be proven safe to flip, so the shared `dateTimeT`/`dateTimeSpace` formatters are left untouched. Instead, `CoreFormatters` gains `dateTimeTUTC`/`dateTimeSpaceUTC` (UTC + `en_US_POSIX`) and a `parseDateTimeUTC()` cascade, and `BadgeCountService` (`hasOldItems` + the `oldestPendingDate` min-selection) now uses that UTC path, since its zone-less inputs are exclusively SQLite-generated `datetime('now')` / `CURRENT_TIMESTAMP` values which are always UTC. Two regression tests added: a deterministic epoch-mapping test and 7-day-threshold boundary cases (7d2h/6d22h) — verified the boundary test fails against the pre-fix local-tz parser on this UTC-6 machine and passes with the UTC path.
